### ISSUE TITLE
Change default app for opening downloaded files

### ISSUE THREAD

**Issue by Cwpute** [Slion/Fulguris#584]
**Is your feature request related to a problem? Please describe.**
Some day after a file download, i mistapped on the wrong app when asked what i wanted to open that file with. Since then, i have been unable to change that default app back to the one i wanted.
This is not a system wide problem, as pdf i open through my file browser are correctly picked up by my pdf viewer.

**Describe the solution you'd like**
Some way to reset default apps, or even set them manually in the settings.

**Comment by Slion** [comment]
I'm pretty sure it is a system option. However some devices don't let you reset it, it seems. At least I could not find a way to reset it on my Galaxy Tab S8 Ultra.

**Comment by Cwpute** [comment]
I don't have that option available in my device settings unfortunately.
Reinstalling Fulguris is what i'll end up doing if i don't have any alternative soon :/

**Comment by Slion** [comment]
@Cwpute I don't think reinstalling Fulguris will help. What meant above is uninstall the app you chose as default file browser. If possible.

**Comment by gabrielparca** [comment]
@Cwpute I feel your pain... it's bonkers that on our own phones (I own a Gallaxy, btw) we aren't able to choose which app opens a certain file extension—or even a category filetype.

The One UI 6 (I have migrated recently—couple of months—from Huawei, so dk about previous versions) just let's choose default apps for half a dozen more operational things like browser, phone, sms, assistant, launcher... 

@Slion remarked, reinstalling Fullguris wouldn't help, as when you download a file on an Android phone, the browser used to download it doesn’t decide which app opens it. It's the system that has the list of apps that are set as defaults for different file types—which the browser will automatically use to open the file. If not, it'll do as you said and present a list of options to choose from. Upon choosing, this then becomes automatic. You may try whatever app to open those same downloaded files, and you will invariably get the app you have mistapped on initially. 

As you cannot change this, the thing would be what @slion advised and uninstall (reinstall) to be able to choose again. But before that, you would want to check if you can go to that app's info and clear its defaults, and if so you're good to go without any of those extra steps.

**Comment by Slion** [comment]
> I feel your pain... it's bonkers that on our own phones (I own a Gallaxy, btw) we aren't able to choose which app opens a certain file extension—or even a category filetype.

It's Android, I would bet there are third party apps out there that let you fix that somehow.

**Comment by Slion** [comment]
> As you cannot change this, the thing would be what @Slion advised and uninstall (reinstall) to be able to choose again. But before that, you would want to check if you can go to that app's info and clear its defaults, and if so you're good to go without any of those extra steps.

Some phones/launchers let you disable apps without uninstalling them. That's probably even easier than uninstalling.

**Comment by Slion** [comment]
I wonder if the Nuclear option mentioned [there](https://www.androidauthority.com/change-default-apps-android-2-633572/) might work for you. Far from ideal though as it surely would mess up a bunch of other things too.

Found that [Default App Manager](https://play.google.com/store/apps/details?id=com.tintage.android.defaultappmanager), did not try it as I'm assuming it's not working anymore.

**Comment by Slion** [comment]
Also you may want to install [App Manager](https://github.com/MuntashirAkon/AppManager) from F-Droid. I don't think it features something to manage default apps but I wonder if simply installing it somehow forces the pick an app dialog.

**Comment by Cwpute** [comment]
So after some time i didn't need naymore the app i faultily selected as default for opening downloads in Fuguris, so i uninstalled it. Everything is back to normal. Although i believe this is still a problem that should find a fix.

**Comment by Slion** [comment]
Definitely an Android default app settings issue. You can clear them by going to the system settings for your app and clear defaults.

**Comment by Cwpute** [comment]
After i posted my last message, some time passed and i reinstalled the app that was automatically opening downloaded files. And Fulguris immediately started opening downloaded files in that app, although i hadn't told it to default to anything. Do you think that it's still an Android default app problem ?

**Comment by Slion** [comment]
Which app are you talking about? I would need to take a closer look at the code. If I recall we might be prioritizing third-party apps over built-in ones. Making the assumption that if you install a third-party file explorer you actually want to use it over the built-in ones.

Ideally there should be a setting option for user to choose the file explorer they want Fulguris to use.

**Comment by Cwpute** [comment]
It is not a file explorer : https://github.com/WirelessAlien/ZipXtract
Maybe ZipXtract is faking itself as a file explorer for some reason, which might explain why it's always picking it as the default app. (i don't have any other file explorer except from the default android one)

**Comment by Slion** [comment]
Does it not display the download folder then?

**Comment by Cwpute** [comment]
No, it shows the app window for entering a password, as if it was opening a archive file.

**Comment by Slion** [comment]
Thanks so I tried it out and in my case since I don't have a default set in system settings it asks me which app to use including that ZipXtract. So it's really a case of the app telling the system it can show folders but does not. It's also caused by the fact that Fulguris first tries this third-party app way of showing folders before trying the system built-in way if no third-party found. In your case you only have one third-party app installed supporting that feature so it just uses that one without asking which one to use as default first.

As a workaround you could install another third-party app. I use File Manager + but I don't think it's free. You could try App Manager I guess: https://github.com/MuntashirAkon/AppManager

I guess we could report this as a bug to the developer of ZipXtract see what they think and if they really need that feature they expose. I'm not sure I'll bother though.

![Image](https://github.com/user-attachments/assets/c6e03f05-1b59-4fc8-80c9-e0d05075c504)

On that screenshot you see the upcoming Fulguris download list #758.

**Comment by Cwpute** [comment]
i bothered :)

**Comment by Slion** [comment]
There must be a way to clear the defaults though, here are screenshots from Magic OS for instance:

![Image](https://github.com/user-attachments/assets/fd91f9ce-ce08-4519-b4e0-63c3afe63a41)

![Image](https://github.com/user-attachments/assets/82de4f3e-2750-4921-94c5-84b7439b505e)

**Comment by Slion** [comment]
@Cwpute Did you find the clear defaults option on your phone?

**Comment by Cwpute** [comment]
Yes i did, and it seems to have done the trick. Weird, because i remember having done this and it not working before.
Anyway, now the problem is different, when i try to open a downloaded file from Fulguris' popup when the download is finished, it won't even show me an app picker but just tells me "No file selected" in a pop-up.

**Comment by Slion** [comment]
That's odd, what do you get when using Downloads menu item?

**Comment by Cwpute** [comment]
I get a glimpse of a dark window with a progress bar that says "Extracting…" …and then the same "No file selected popup" comes up.
So it did change something, but apparently ZipXtract is still tied to opening downloaded files.

**Comment by Slion** [comment]
What's your phone make, model and Android version?

**Comment by Cwpute** [comment]
Nokia TA-1157, Android 11

**Comment by Slion** [comment]
@Cwpute Thanks for your collaboration, the issue has been fixed in ZipXtract v7.1.
I've opened another related issue #766 and we are done for good here, hopefully :)

**Comment by Cwpute** [comment]
it did fix the issue for me ! at last :D thank you as well for your patience and cooperation.
